### ISSUE TITLE
Mean/max pooling variants bypass the embedding layer is my understanding correct?

### ISSUE THREAD

**Issue by Geongyu** [naivete5656/CPNN#2]
Hi, thanks for releasing the code — it has been very helpful for our benchmark.

While reproducing the comparison methods, I noticed something about the mean and max variants of AbMIL and wanted to check whether I am understanding the implementation correctly.

In model/comparisons/abmil.py, the attention-based variant passes the input features through the learned projection:

self.attention_net = nn.Sequential(
    nn.Linear(size[0], size[1]),
    nn.ReLU(),
    attention_net
)

and in forward:

A, h = self.attention_net(h)
M = torch.mm(F.softmax(A.transpose(1, 0), dim=1), h)

However, for the mean and max variants, pooling seems to be applied directly to the input features:

elif "mean" in self.version:
    M = h.mean(0).unsqueeze(0)

elif "max" in self.version:
    M = h.max(0)[0].unsqueeze(0)

and the prediction head is correspondingly changed to:

self.head = nn.Linear(size[0], n_classes)

So, if I understand correctly, the mean and max variants do not use the Linear(size[0], size[1]) + ReLU embedding used by the attention variant. With pre-extracted/frozen CONCH features, these models therefore reduce to mean/max pooling followed by a linear prediction head.

Is this interpretation correct, and was this behavior intentional?

I am asking because, in our experiments, keeping the same learned embedding before mean/max pooling noticeably changed the results. For example, on TCGA-BRCA with the same patient-level 4-fold protocol:

Mean pooling: Spearman 0.312 → 0.338
Max pooling: Spearman 0.202 → 0.283

We observed the same direction on TCGA-KIRC and TCGA-LUAD as well.

I mainly wanted to confirm whether the current mean/max implementation is intended to operate directly on the raw CONCH features, rather than using the same embedding layer as the attention variant.

Thanks!
